### PR TITLE
fix vulns

### PR DESCRIPTION
**Pull request by ryhuz**
**Changes**
- upgrade to storybook 8
  - removed autodocs since all stories have their attached `.mdx` and is of a different layout to the autodocs
- bump postcss-import to v16
  - add postcss@v8 as a devDependency since it is now a peer-dep of postcss-import
  - this fixes the vuln in postcss@v7

- [delete] branch

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,4 +16,5 @@ coverage
 
 storybook-static
 
-junit.xml
\ No newline at end of file
+junit.xml
+*storybook.log
```
Comment (qroll) [inline .gitignore:20]: what generates this file?

File: .storybook/main.ts [hunk 2]
```diff
@@ -33,14 +24,24 @@ module.exports = {
...
         return config;
     },
-    framework: "@storybook/react-webpack5",
-    docs: {
-        autodocs: true,
+    framework: {
+        name: "@storybook/react-webpack5",
+        options: {},
     },
+    swc: () => ({
```
Comment (qroll) [inline .storybook/main.ts:37]: can explain what this option does?

File: .storybook/main.ts [hunk 3]
```diff
@@ -1,26 +1,17 @@
...
 
-module.exports = {
-    stories: [
-        "../stories/**/!(*.stories).mdx",
-        "../stories/**/*.stories.@(ts|tsx)",
-    ],
+const config: StorybookConfig = {
+    stories: ["../stories/**/*.mdx", "../stories/**/*.stories.@(ts|tsx)"],
     addons: [
+        "@storybook/addon-webpack5-compiler-swc",
```
Comment (qroll) [inline .storybook/main.ts:8]: what does this addon do?

File: .storybook/main.ts [hunk 4]
```diff
@@ -1,26 +1,17 @@
...
         "@storybook/addon-links",
         "@storybook/addon-essentials",
         "@storybook/addon-a11y",
         "@storybook/addon-themes",
-        {
-            name: "@storybook/addon-docs",
-            options: {
-                mdxPluginOptions: {
-                    mdxCompileOptions: {
-                        remarkPlugins: [remarkGfm],
```
Comment (qroll) [inline .storybook/main.ts:19]: this was needed to handle markdown for a specific syntax, but I can't remember what exactly

**Comment by qroll** [comment]
the intro documentation for `Text` seems to be missing

https://designsystem.life.gov.sg/react/index.html?path=/docs/general-text-introduction--docs

File: .storybook/main.ts (hunk 2 above)
Comment (ryhuz) [reply .storybook/main.ts:37]: the swc option was needed for the storybook to work. the compiler options out of the box didn't work. got this from stack overflow

File: .storybook/main.ts (hunk 3 above)
Comment (ryhuz) [reply .storybook/main.ts:8]: with a fresh init of storybook, this was the compiler set up. can't rmb which option this was but there were 2 suggested..

File: .gitignore (hunk 1 above)
Comment (ryhuz) [reply .gitignore:20]: this line you mean? i did a fresh init of storybook.

**Comment by ryhuz** [comment]
> the intro documentation for `Text` seems to be missing
> 
> https://designsystem.life.gov.sg/react/index.html?path=/docs/general-text-introduction--docs



> the intro documentation for `Text` seems to be missing
> 
> https://designsystem.life.gov.sg/react/index.html?path=/docs/general-text-introduction--docs

🤔 let me check. none of the stories were changed.

File: .storybook/main.ts (hunk 4 above)
Comment (ryhuz) [reply .storybook/main.ts:19]: image
yes gets this error

**Comment by ryhuz** [comment]
> the intro documentation for `Text` seems to be missing
> 
> https://designsystem.life.gov.sg/react/index.html?path=/docs/general-text-introduction--docs

found the issue. i think the new <Meta> needs a component? i'm guessing. but if i hardcode the Meta title in the `.mdx` then it's ok

File: .storybook/main.ts (hunk 4 above)
Comment (qroll) [reply .storybook/main.ts:19]: ok figured out, it's required for tables to work in mdx http://localhost:6006/?path=/docs/modules-errordisplay--docs

is there a new version that's compatible with SB8?

Comment (ryhuz) [reply .storybook/main.ts:19]: seems like the v3 -> v4 fixes this

**Review by qroll — APPROVED** [review]

**Pull request merged**